client/src/main/java/cloud/prefab/client/ConfigClient.java

```diff
@@ -25,6 +25,8 @@ Optional<Prefab.ConfigValue> get(
     Map<String, Prefab.ConfigValue> properties
   );
 
+  public Map<String, Prefab.ConfigValue> getAllValues();
```

Comment:
I realized that `Prefab.ConfigValue` is just a protobuf so it won't pick up any updates going forward. I wonder if we should try to find a pattern that encourages the use of live values

Reply:
The intended usage afaik only requires the Type. It could be ‘Map<String, Config.Type>’  if we want to make it smaller surface area. 

Or yeah, something that is live would be nice.

Reply:
You're right, exposing the unchanging value directly is a bit of a foot gun. I'm not sure how to dole out live values here though given the type params and type erasure unless its `Key -> wrapperClass{type, Supplier<Type>}`

Reply:
We could maybe have a method like `Set<ConfigSupplier<?>> getAllSuppliers()` where `ConfigSupplier` is an object/interface like:
```java
String getKey();
Supplier<?> getValueSupplier();
Type getSupplierGenericType();
```

And then users can do something like:
```java
for (ConfigSupplier<?> configSupplier : configClient.getAllSuppliers()) {
  binder
    .bind(configSupplier.getSupplierGenericType(), Config.named(configSupplier.getKey())
    .toInstance(configSupplier.getValueSupplier());
}
```

Reply:
Is there a good way to convey `configSupplier.getSupplierGenericType()` without requiring guice's TypeLiteral? We want the client to be injection framework agnostic

Reply:
The JDK has a `ParameterizedType` interface, and this method would return an instance of that. However, the JDK doesn't have a built-in implementation of the interface which is a bit annoying (pretty simple to implement ourselves though)